### ISSUE TITLE
"Connections.uniqueRoot" not specified in Modelica 3.2

### ISSUE THREAD

**Issue by modelica-trac-importer** [HansOlsson/Modelica_StateGraph2#2]
**Reported by Brunnengräber on 8 Feb 2013 21:47 UTC**
why does stategraph2 use a not specified function?

how is "Connections.uniqueRoot" defined and how can it get implemented into the modelica 3.2 specification?

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/984

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 30 Mar 2013 13:14 UTC**
In 2009 there was a proposal to add two new language elements to the coming Modelica 3.2 specification. The proposal is attached in the attachment above (Modelica_Proposal_StateDiagrams.pdf). These two language elements are needed to implement the Modelica_StateGraph2 library (and they are supported by Dymola). An explanation why they are needed is given in the appendix of the paper http://www.ep.liu.se/ecp/043/041/ecp09430108.pdf. However, when releasing Modelica Specification 3.2 it was decided to not include this proposal. The reason is that it was felt that Modelica_StateGraph2 was not yet good enough for state machine description in Modelica. Instead in Modelica 3.3 state machines have been introduced in Modelica by additional language elements to model Lucid Synchrone type of state machines. The state machines in Modelica 3.3 are only for sampled data systems. Currently work is performed in the MODRIO project to generalize this also to continuous-time state machines.

To summarize: Modelica_LinearSystems2 is not fully Modelica compliant and will never be, since a better solution is now available with Modelica 3.3.

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 1 Apr 2013 12:17 UTC**
Just a tiny correction. I think Martin Otter meant to say:

Replying to [comment:1 otter]:
> To summarize: ~~Modelica_LinearSystems2~~ Modelica_StateGraph2 is not fully Modelica compliant and will never be, since a better solution is now available with Modelica 3.3.

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 12 May 2013 22:38 UTC**
